c/parallel/src/transform.cu

```diff
@@ -0,0 +1,617 @@
+      input_it.value_type.size, // 0
+      input_it.value_type.alignment, // 1
+      output_it.value_type.size, // 2
+      output_it.value_type.alignment, // 3
+      policy.block_threads, // 4
+      policy.items_per_thread_no_input, // 5
+      policy.min_items_per_thread, // 6
+      policy.max_items_per_thread, // 7
+      input_iterator_src, // 8
+      output_iterator_src, // 9
+      op_src); // 10
+
+#if false // CCCL_DEBUGGING_SWITCH
+    fflush(stderr);
+    printf("\nCODE4NVRTC BEGIN\n%sCODE4NVRTC END\n", src.c_str());
+    fflush(stdout);
+#endif
+
+    std::string kernel_name = transform::get_kernel_name(input_it, output_it, op);
+    std::string kernel_lowered_name;
+
+    const std::string arch = std::format("-arch=sm_{0}{1}", cc_major, cc_minor);
+
+    constexpr size_t num_args  = 8;
+    const char* args[num_args] = {
+      arch.c_str(), cub_path, thrust_path, libcudacxx_path, ctk_path, "-rdc=true", "-dlto", "-default-device"};
+
+    constexpr size_t num_lto_args   = 2;
+    const char* lopts[num_lto_args] = {"-lto", arch.c_str()};
+
+    // Collect all LTO-IRs to be linked.
+    nvrtc_ltoir_list ltoir_list;
+    nvrtc_ltoir_list_appender appender{ltoir_list};
+
+    appender.append({op.ltoir, op.ltoir_size});
+    appender.add_iterator_definition(input_it);
+    appender.add_iterator_definition(output_it);
+
+    nvrtc_link_result result =
+      make_nvrtc_command_list()
+        .add_program(nvrtc_translation_unit{src.c_str(), name})
+        .add_expression({kernel_name})
+        .compile_program({args, num_args})
+        .get_name({kernel_name, kernel_lowered_name})
+        .cleanup_program()
+        .add_link_list(ltoir_list)
+        .finalize_program(num_lto_args, lopts);
+
+    cuLibraryLoadData(&build_ptr->library, result.data.get(), nullptr, nullptr, 0, nullptr, nullptr, 0);
+    check(cuLibraryGetKernel(&build_ptr->transform_kernel, build_ptr->library, kernel_lowered_name.c_str()));
+
+    build_ptr->loaded_bytes_per_iteration = input_it.value_type.size;
+    build_ptr->cc                         = cc;
+    build_ptr->cubin                      = (void*) result.data.release();
+    build_ptr->cubin_size                 = result.size;
+  }
+  catch (const std::exception& exc)
+  {
+    fflush(stderr);
+    printf("\nEXCEPTION in cccl_device_transform_build(): %s\n", exc.what());
+    fflush(stdout);
+    error = CUDA_ERROR_UNKNOWN;
+  }
+
+  return error;
+}
+
+CUresult cccl_device_unary_transform(
+  cccl_device_transform_build_result_t build,
+  cccl_iterator_t d_in,
+  cccl_iterator_t d_out,
+  unsigned long long num_items,
+  cccl_op_t op,
+  CUstream stream)
+{
+  bool pushed    = false;
+  CUresult error = CUDA_SUCCESS;
+  try
+  {
+    pushed = try_push_context();
+
+    CUdevice cu_device;
+    check(cuCtxGetDevice(&cu_device));
+    auto cuda_error = cub::detail::transform::dispatch_t<
+      cub::detail::transform::requires_stable_address::no, // TODO implement yes
+      ::cuda::std::int64_t,
```

Comment:
Should we maybe use `OffsetT` here? I know this is not how the other algorithms do it, but I think it makes more sense